pkg/util/cache/cache.go

```diff
@@ -27,6 +27,9 @@ const (
 type Cache []*cacheShard
 
 func NewCache(maxSize int) Cache {
+	if maxSize < shardsCount {
```

Comment:
I agree with @lavalamp here. What we should really do is to set number of shards to min(shardsCount, maxSize) here